tests/test_all_rules.py

```diff
@@ -292,6 +292,48 @@ def test_required_tags(self):
             if missing_required_tags or is_missing_any_tags:
                 self.fail(error_msg)
 
+    def test_primary_tactic_as_tag(self):
+        from detection_rules.attack import tactics
+
+        invalid = []
+        tactics = set(tactics)
+
+        for rule in self.rules:
+            rule_tags = rule.contents['tags']
+
+            if 'Continuous Monitoring' in rule_tags or rule.type == 'machine_learning':
+                continue
+
+            threat = rule.contents.get('threat')
+            if threat:
+                missing = []
+                threat_tactic_names = [e['tactic']['name'] for e in threat]
+                primary_tactic = threat_tactic_names[0]
+
+                if 'Threat Detection' not in rule_tags:
+                    missing.append('Threat Detection')
+
+                # missing primary tactic
+                if primary_tactic not in rule.contents['tags']:
+                    missing.append(primary_tactic)
+
+                # listed tactic that is not in threat mapping
+                tag_tactics = set(rule_tags) & tactics
```

Comment:
I struggle so much with the binary operators `&` and `|` for sets and all of the methods.
maybe spelling it out is the most clear?
```suggestion
                tag_tactics = set(rule_tags).intersection(tactics)
```